gtsfm/evaluation/metrics.py

```diff
@@ -0,0 +1,362 @@
+                    self._plot_type = self.PlotType.HISTOGRAM if "histogram" in summary else self.PlotType.BOX
+                else:
+                    self._plot_type = plot_types_for_dim[0]
+            elif plot_type in plot_types_for_dim:
+                self._plot_type = plot_type
+            else:
+                raise ValueError("Unsupported plot type for the data dimension")
+
+            # Create a summary if the data is a 1D distribution
+            if self._dim == 1:
+                self._summary = self._create_summary(data)
+
+            # Store full data only if its a scalar or if asked to.
+            if self._dim == 0 or store_full_data:
+                self._data = data
+            else:
+                self._data = None
+        else:
+            self._dim = 1
+            self._summary = summary
+            self._plot_type = self.PlotType.HISTOGRAM if "histogram" in summary else self.PlotType.BOX
+            self._data = None
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def data(self) -> np.array:
+        return self._data
+
+    @property
+    def plot_type(self) -> PlotType:
+        return self._plot_type
+
+    @property
+    def dim(self) -> int:
+        return self._dim
+
+    @property
+    def summary(self) -> Dict[str, Any]:
+        return self._summary
+
+    def _get_plot_types_for_dim(self, dim: int) -> List[PlotType]:
+        if dim == 0:
+            return [self.PlotType.BAR]
+        if dim == 1:
+            return [self.PlotType.BOX, self.PlotType.HISTOGRAM]
+        return []
+
+    def _get_distribution_histogram(self, data: np.ndarray) -> Dict[str, Union[float, int]]:
+        """Returns the histogram of data as a dictionary.
+
+        If the data is float, the keys of the dictionary are interval buckets, and if the data is int, the keys are also int.
+
+        Args:
+            data: 1D array of all values of the metric
+
+        Returns:
+            Histogram of data as a dict from bucket to count.
+        """
+        if data.size == 0:
+            logger.info("Requested histogram for empty data metric, returning None.")
+            return None
+        if isinstance(data.tolist()[0], int):
+            # One bin for each integer
+            bins = int(np.max(data) - np.min(data) + 1)
+            discrete = True
+        else:
+            bins = 10
+            discrete = False
+        count, bins = np.histogram(data, bins=bins)
+        count = count.tolist()
+        bins = bins.tolist()
+        bins_lower = bins[:-1]
+        bins_upper = bins[1:]
+
+        histogram = {}
+        for i in range(len(count)):
+            if discrete:
+                key = str(int(bins_lower[i]))
+            else:
+                key = "%.2f-%.2f" % (bins_lower[i], bins_upper[i])
+            histogram[key] = count[i]
+        return histogram
+
+    def _create_summary(self, data: np.ndarray) -> Dict[str, Any]:
```

Comment:
same comment here. Does not need to be in the class.

Reply:
i think i prefer it in the class, since it references the PlotTypes, which are part of the class?

Reply:
prefer not a free function